### ISSUE TITLE
Longwave radiation is underestimated 

### ISSUE THREAD

**Issue by arbennett** [UW-Hydro/MetSim#16]
There is ~4% consistent underestimation of longwave radiation that I've been unable to track down fully.  Below is a plot showing this:  

![lw](https://cloud.githubusercontent.com/assets/7518706/22090519/0338fdb8-dda7-11e6-85b9-3eb041ab2174.png)

The red is the output of VIC4.2 and the blue underneath is the output of MetSim.  There are only 3 inputs to the longwave calculation: temperature, vapor pressure, and cloud fraction.  Both temperature and cloud fraction have deviations from VIC4.2 of <0.1%, and don't show a consistent bias direction for those deviations, so it seems they are not to blame.  Similarly, the vapor pressure doesn't show a consistent bias, so I don't think it is the cause either.  This leaves the calculation of the longwave radiation itself to be the problem.  The code to do so is:

```
def longwave(air_temp, vapor_pressure, tskc):
    """ Calculate longwave """
    emissivity_calc = {
            'DEFAULT'    : lambda x : x,
            'TVA'        : lambda x : 0.74 + 0.0049 * x,
            'ANDERSON'   : lambda x : 0.68 + 0.036 * np.power(x, 0.5),
            'BRUTSAERT'  : lambda x : 1.24 * np.power(x/air_temp, 0.14285714),
            'SATTERLUND' : lambda x : 1.08 * (1 - np.exp(-1 * np.power(x, (air_temp/2016)))),
            'IDSO'       : lambda x : 0.7 + 5.95e-5 * x * np.exp(1500/air_temp),
            'PRATA'      : lambda x : (1 - (1 + (46.5*x/air_temp)) *
                np.exp(-np.sqrt((1.2 + 3. * (46.5*x/air_temp)))))
            }
    cloud_calc = {
            'DEFAULT' : lambda x : (1.0 + (0.17 * tskc**2)) * x,
            'CLOUD_DEARDORFF' : lambda x : tskc + (1-tskc)*x
            }
    # Reindex and fill cloud cover, then convert temps to K
    tskc = tskc.reindex(air_temp.index)
    tskc = tskc.fillna(method='ffill')
    air_temp = air_temp + conf.KELVIN 

    # Calculate longwave radiation based on the options 
    emissivity_clear = emissivity_calc[conf.LW_TYPE.upper()](vapor_pressure)
    emissivity = cloud_calc[conf.LW_CLOUD.upper()](emissivity_clear) 
    lwrad = emissivity * conf.STEFAN_B * np.power(air_temp, 4)
    return lwrad, tskc
```

This code is relatively simple, and uses the `PRATA` and `CLOUD_DEARDORFF` options.  It _ought_ to just be finding the typo somewhere, but I've yet to find one.  Note that the error is not constant across the time, so it's not just a scale factor somewhere.

**Comment by arbennett** [comment]
Here's another plot, but zoomed in so that it's more apparent how the error really looks.

![lw_zoom](https://cloud.githubusercontent.com/assets/7518706/22091001/5571fa88-dda9-11e6-965e-43ad34d5bf65.png)

**Comment by bartnijssen** [comment]
The VIC function `calc_longwave()` (https://github.com/UW-Hydro/VIC/blob/support/VIC.4.2.d/src/calc_longwave.c#L35) starts with a unit conversion for the air temperature and for `vp` (`vp /= 100; // convert to mbar`). You do the unit conversion for `air_temp`, but are you sure `vp` is in the correct units?

**Comment by bartnijssen** [comment]
Also:

https://github.com/UW-Hydro/VIC/blob/support/VIC.4.2.d/src/vicNl_def.h#L289 : `#define STEFAN_B     5.6696e-8	/* stefan-boltzmann const in unit W/m^2/K^4 */`

https://github.com/UW-Hydro/MetSim/blob/master/metsim/defaults.py#L30 : `'STEFAN_B' : 5.669e-8  # (W m^-22 K^-4) Stefan Boltzmann constant ` (I assume that this is where your `conf.STEFAN_B` comes from - if it actually comes from a configuration file, please check that although this should really be a physical constant)

The difference does not seem large enough to explain the discrepancy since it is linear in the constant and the ratio between the two is 0.99989 (with yours lower), but it's better to keep the constants identical.

**Comment by bartnijssen** [comment]
Actually, can you plot the ratio so we can see whether the problem is in the scalar multiplier (e.g. the Stefan Boltzmann constant) or elsewhere in the formulation.

**Comment by arbennett** [comment]
Got it!  It was a unit conversion error, that I introduced when calculating the vapor pressure within the `mtclim` module.  In VIC the vapor pressure unit conversion happened somewhere else, and I moved it into `mtclim`  for simplicity, see #14 for that discussion.

Here's "proof" of the working example:

![lw_works](https://cloud.githubusercontent.com/assets/7518706/22092792/6dc195ce-ddb5-11e6-99a8-50627536f2d9.png)

And a side note: Right now all physical constants are stored in `configuration.py`.  It will be updated to be `constants.py` and then imported as `consts` instead, for clarity.
